lib/cosmos/config/config_parser.rb

```diff
@@ -347,30 +337,28 @@ def self.handle_defined_constants(value, data_type = nil, bit_size = nil)
 
     protected
 
-    # Writes the parsed results for debugging if we had an error parsing
-    def create_debug_output_file(filename, file, unparsed_data, exception)
+    # Writes the ERB parsed results
+    def create_parsed_output_file(filename)
       begin
-        debug_file = nil
-        tmp_folder = File.join(Cosmos::USERPATH, 'outputs', 'tmp')
-        tmp_folder = Cosmos::USERPATH unless File.exist?(tmp_folder)
-        debug_file = File.join(tmp_folder, "parser_error_#{File.basename(filename)}")
-        File.open(debug_file, 'w') do |save_file|
-          save_file.puts exception.formatted
-          save_file.puts "\nParsed Data (will only be present if parse ran successfully):"
-          save_file.puts
-          if file
-            file.rewind
-            save_file.puts file.read
-          end
-          save_file.puts "\nUnparsed Data:"
-          save_file.puts
-          save_file.puts unparsed_data if unparsed_data
-        end
-      rescue Exception
-        # Oh well - we tried
-        debug_file = nil
+        output = ERB.new(File.read(filename)).result(binding)
+      rescue => e
+        # The first line of the backtrace indicates the line where the ERB
+        # parse failed. Grab the line number for the error message.
+        match = /:(.*):/.match(e.backtrace[0])
+        line_number = match.captures[0] if match
+        raise e, "ERB error at #{filename}:#{line_number}\n#{e}", e.backtrace
       end
-      debug_file
+
+      # Make a copy of the filename since we're calling slice! which modifies
+      # it directly. Downcase to avoid filesytem case issues.
+      copy = filename.downcase.dup
+      copy.slice!(Cosmos::USERPATH.downcase) # Remove the USERPATH
```

Comment:
Should check if the filename was actually in the USERPATH.  It almost always will be, but what if it isn't?